src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -73,14 +73,51 @@ let non_pc_registers_equal_var t1 t2 =
       let ( ! ) eq x1 x2 = Impl.run_checked (eq x1 x2) in
       let f eq acc field = eq (F.get field t1) (F.get field t2) :: acc in
       Registers.Fields.fold ~init:[]
-        ~ledger:(f !Frozen_ledger_hash.equal_var)
+        ~first_pass_ledger:(f !Frozen_ledger_hash.equal_var)
+        ~second_pass_ledger:(f !Frozen_ledger_hash.equal_var)
         ~pending_coinbase_stack:(fun acc f ->
           let () = F.get f t1 and () = F.get f t2 in
           acc )
         ~local_state:(fun acc f ->
           Local_state.Checked.equal' (F.get f t1) (F.get f t2) @ acc )
       |> Impl.Boolean.all )
 
+let txn_statement_ledger_hashes_equal
+    (s1 : Transaction_snark.Statement.With_sok.Checked.t)
+    (s2 : Transaction_snark.Statement.With_sok.Checked.t) =
+  Impl.make_checked (fun () ->
+      let module F = Core_kernel.Field in
+      let ( ! ) x = Impl.run_checked x in
+      let source_eq =
+        !(non_pc_registers_equal_var
+            { s1.source with pending_coinbase_stack = () }
```

Comment:
if the `pending_coinbase` parameter of this type is `unit`, which has one possible value, there's no need to replace that field

if omitting the field update leads to an ambiguous type, better to specify the complete type than use the field update

ditto for the other occurrences of this mechanism below

Reply:
`pending_coinbase_stack` in `s1` and `s2` has type `Pending_coinbase.Stack.var`